scripts/o3de/o3de/manifest.py

```diff
@@ -301,6 +301,83 @@ def get_engine_templates() -> list:
 def get_project_gems(project_path: pathlib.Path) -> list:
     return get_gems_from_external_subdirectories(get_project_external_subdirectories(project_path))
 
+def get_project_enabled_gems(project_path: pathlib.Path, include_dependencies:bool = True) -> dict or None:
+    """
+    Returns a dictionary of "<gem name with optional specifier>":"<gem path>"
+    Example: {"gemA>=1.2.3":"c:/gemA", "gemB":"c:/gemB"}
+    :param project_path The path to the project
+    :param include_gem_dependencies True to include all gem dependencies, otherwise just return
+    gems listed in project.json and the deprecated enabled_gems.json
+    """
+    project_json_data = get_project_json_data(project_path=project_path)
+    active_gem_names = project_json_data.get('gem_names',[])
+    enabled_gems_file = cmake.get_enabled_gem_cmake_file(project_path=project_path)
+    if enabled_gems_file and enabled_gems_file.is_file():
+        active_gem_names.extend(cmake.get_enabled_gems(enabled_gems_file))
+
+    gem_names_with_optional_gems = utils.get_gem_names_set(active_gem_names, include_optional=True)
+    if not gem_names_with_optional_gems:
+        return {}
+    
+    # We have the gem names but not the resolved paths yet
+    result = {gem_name: None for gem_name in gem_names_with_optional_gems}
+
+    engine_path = get_project_engine_path(project_path=project_path)
+    if not engine_path:
+        engine_path = get_this_engine_path()
+        if not engine_path:
+            logger.error('Failed to find an engine path for the project at '
+                            f'"{project_path}" which is required to resolve gem dependencies.')
+            return result
+
+        logger.warning('Failed to determine the correct engine for the project at '
+                        f'"{project_path}", falling back to this engine at {engine_path}.')
```

Comment:
Does `get_proejct_engine_path` take into the actual path the project is using?
Such as if their `user/project.json` is set to a different engine?

Reply:
yes, it calls `get_most_compatible_project_engine_path` which takes `user/project.json` into account.